### PR TITLE
[Text Generation] [Fix] Raise error when we use deepsparse engine and `prompt_processing_length == sequence_lenght`

### PR DESCRIPTION
**Pull request by dbogunowicz**
When we try to create a multi-token deepsparse engine where `prompt_processing_length == sequence_length`, this engine will default to onnxruntime:
```bash
[nm_ort 7f9cd3579740 >WARN<  is_supported_graph src/onnxruntime_neuralmagic/supported/ops.cc:144] Warning: Optimized runtime disabled - Detected dynamic input past_key_values.0.key dim 2. Set inputs to static shapes to enable optimal performance.
```
this is a known issue reported by the runtime team. The crux is: if `prompt_processing_length == sequence_length`, the initial kv cache tensors will be empty, and this is prohibitive in the engine - we cannot have empty kv cache values in there.

### CODE REVIEW

**Comment by rsnm2** [comment]
@dbogunowicz 

This makes sense.

However, the eval_downstream script sets the prompt_prefill and sequence_length equal to the same value. So now running eval_downstream will fail with your error message

https://github.com/neuralmagic/deepsparse/blob/main/src/deepsparse/transformers/eval_downstream.py#L89

**Comment by dbogunowicz** [comment]
@rsnm2 yeah, missing commit. fixed :)

**Comment by rahul-tuli** [comment]
Note: failing tests should be fixed by https://github.com/neuralmagic/sparsezoo/pull/357

**Review by bfineran — DISMISSED** [review]

File: src/deepsparse/transformers/pipelines/text_generation.py [hunk 1]
```diff
@@ -214,6 +214,20 @@ def initialize_engines(
...
+                self.engine_type == DEEPSPARSE_ENGINE
+                and self.sequence_length <= self.prompt_processing_sequence_length
+                and self.enable_multitoken_prefill
+            ):
+                raise ValueError(
+                    "Attempting to initialize auxiliary DeepSparse engine to "
+                    "process a prompt with a larger processing length. "
+                    "However, it is assumed that `prompt_processing_sequence_length` "
+                    "is smaller than the `sequence_length`. "
+                    "Please adjust the `prompt_processing_sequence_length` "
```
Comment (dbogunowicz) [inline src/deepsparse/transformers/pipelines/text_generation.py:227]:
```suggestion
                    "Adjust the `prompt_processing_sequence_length` "
```

**Review by Satrat — APPROVED** [review]

**Review by rahul-tuli — APPROVED** [review]

**Pull request merged**